src/form.tsx

```diff
@@ -181,16 +196,49 @@ type FormState<Data extends Record<string, unknown> = Record<string, unknown>> =
  * looking loading status.
  */
 export const Form = forwardRef<HTMLFormElement, FormProps>(function Form(
-  { method = 'post', onSubmit, ...props },
+  { method = 'post', onSubmit, onSuccess, onError, ...props },
   forwardRef,
 ) {
   const router = useRouter();
   const ref = useRefs<HTMLFormElement>(forwardRef);
 
   const name = props.name || UNNAMED_FORM;
+
   const set = useStore(useCallback((store) => store.set, []));
   const state = useStore(useCallback((store) => store.get(name), [name]));
 
+  // the effects need to check if this is reset to idle
+  // before calling success/error callbacks
+  const initializedRef = useRef(false);
+
+  const onSuccessRef = useLatestRef(onSuccess);
+  useEffect(() => {
+    if (!initializedRef.current) return;
+    if (state.status === 'success') {
+      onSuccessRef.current?.(state);
+    }
+  }, [onSuccessRef, state]);
+
+  const onErrorRef = useLatestRef(onError);
+  useEffect(() => {
+    if (!initializedRef.current) return;
+    if (state.status === 'error') {
+      onErrorRef.current?.(state);
+    }
+  }, [onErrorRef, state]);
+
+  useEffect(() => {
+    // reset form state when we mount
```

Comment:
I'm not sure if we should do that. This means someone can't conditionally show the form without losing state. Think, not showing the firm during loading state, and reverting to open form on error.

Reply:
Maybe we need to add a `reset` method to FormState?

Reply:
Yeah, I initially thought this was needed to ensure the form doesn't immediately call the error/success callback on mount, but that's what the ref is for. I'll remove the state reset